### ISSUE TITLE
Add an empty for OptionT

### ISSUE THREAD

**Issue by DomBlack** [typelevel/cats#935]
Currently OptionT has a `pure` method, but it can be useful to have a `empty` method on the object. At the moment we're simply using the apply to work around it;

```scala
def something(empty: Boolean): OptionT[Future, String] = {
  if (empty) OptionT(Future.success(None)) else OptionT.pure("Hi")
}
```

However `empty` would look cleaner and be easier to understand

```scala
def something(empty: Boolean): OptionT[Future, String] = {
  if (empty) OptionT.empty else OptionT.pure("Hi")
}
```

**Comment by ceedubs** [comment]
:+1: sounds good to me.

It'd look something like this, right?

```scala
def empty[F[_]:Applicative, A]: OptionT[F, A]
```

**Comment by ceedubs** [comment]
I'm wondering if something like `OptionT.none` might be a better name. It might be a little counterintuitive to call `empty[List, String]` and get back a `List` that isn't actually empty (but instead contains a single `None` element). I'm also thinking of `empty[NonemptyList, String]`.

**Comment by adelbertc** [comment]
+1 for `OptionT.none`

**Comment by mikejcurry** [comment]
I think this should be closed after: #946
